File: scala-hoots/src/main/java/io/hoots/fingerprint/domain/FingerPrint.java
package io.hoots.fingerprint.domain;

import io.hoots.input.domain.Item;

import javax.sound.sampled.AudioFormat;
import java.util.List;
import java.util.Map;

/**
 * Created by rwadowski on 14.04.17.
 */
public class FingerPrint {

    private final Map<Hash, List<Point>> print;
    private final Item item;
    private final AudioFormat inputFormat;

    public FingerPrint(Item item, Map<Hash, List<Point>> print, AudioFormat inputFormat) {
        this.item = item;
        this.print = print;
        this.inputFormat = inputFormat;
    }

    public Map<Hash, List<Point>> getPrint() {
        return this.print;
    }

    public Item getItem() {
        return this.item;
    }

    public AudioFormat getInputFormat() {
        return this.inputFormat;
    }
}

File: scala-hoots/src/main/scala/io/hoots/fingerprint/domain/FingerPrint.scala
package io.hoots.fingerprint.domain

import javax.sound.sampled.AudioFormat

import io.hoots.input.domain.Item

/**
  * Created by rwadowski on 19.04.17.
  */
case class FingerPrint(item: Item,
                       print: Map[Hash, Seq[Point]],
                       audioFormat: AudioFormat) {

  def +(c: HashedChunk): FingerPrint = {
    if(print.contains(c.hash)) {
      val points = print(c.hash)
      FingerPrint(item, print + (c.hash -> (points :+ c.point)), audioFormat)
    } else {
      FingerPrint(item, print + (c.hash -> Seq(c.point)), audioFormat)
    }
  }
}
